src/WindowsStateTriggers/EqualsStateTrigger.cs

```diff
@@ -70,27 +70,35 @@ internal static bool AreValuesEqual(object value1, object value2, bool convertTy
 			{
 				return true;
 			}
-			if (value1 != null && value2 != null)
+			if (value1 != null && value2 != null && value1.GetType() != value2.GetType() && convertType)
 			{
-				//Let's see if we can convert - for perf reasons though, try and use the right type in and out
-				if (value1.GetType() != value2.GetType() && convertType)
+				if (ConvertTypeEquals(value1, value2))
 				{
-					var t2 = System.Convert.ChangeType(value1, value2.GetType(), CultureInfo.InvariantCulture);
-					if (value2.Equals(t2))
-					{
-						return true;
-					}
-					//try the other way around
-					t2 = System.Convert.ChangeType(value2, value1.GetType(), CultureInfo.InvariantCulture);
-					if (value1.Equals(t2))
-					{
-						return true;
-					}
+					return true;
+				}
+				// Try the other way around:
+				if (ConvertTypeEquals(value2, value1))
+				{
+					return true;
 				}
 			}
 			return false;
 		}
 
+		private static bool ConvertTypeEquals(object value1, object value2)
+		{
+			// Let's see if we can convert:
+			if (value2 is Enum)
+			{
+				value1 = Enum.ToObject(value2.GetType(), value1);
+			}
+			else
+			{
+				value1 = Convert.ChangeType(value1, value2.GetType(), CultureInfo.InvariantCulture);
```

Comment:
Well the only reason we do type conversion is because when an object type is defined in xaml, it comes back as string.
We really shouldn't be converting anything else. Ie I don't think that 2f == 2d, but the current state this the case.

In any case, throwing is not an option. If you can't convert one type to another, they aren't equal, so return false.

Also the old code tried converting values from value1 to value2, and vice versa. The replaced code doesn't try this.